This adds a missing cast. uintptr_t would have been better here than uint32_t,
but the code is not intended to be portable anyway.
--- a/libgloss/or1k/sbrk.c	2024-05-09 13:38:08.089257549 +0200
+++ b/libgloss/or1k/sbrk.c	2024-05-09 13:35:29.896791467 +0200
@@ -20,7 +20,7 @@
 #include "include/or1k-support.h"
 
 extern uint32_t	end; /* Set by linker.  */
-uint32_t _or1k_heap_start = &end;
+uint32_t _or1k_heap_start = (uint32_t)&end;
 uint32_t _or1k_heap_end;
 
 void *
